### PR TITLE
Wrap `desktopplayback` endpoint with `WithClusterAuth` rather than `WithAuth`

### PR DESCRIPTION
**Pull request by ibeckermayer**
Modifies the desktop playback endpoint so that it uses an `auth.ClientI`
specific to the requested clusterId (aka siteId), rather than always
assuming the root cluster is correct.

### CODE REVIEW

**Review by rosstimothy — APPROVED** [review]

File: lib/web/desktop_playback.go [hunk 1]
```diff
@@ -30,15 +31,21 @@ func (h *Handler) desktopPlaybackHandle(
...
 	p httprouter.Params,
 	ctx *SessionContext,
+	site reversetunnel.RemoteSite,
 ) (interface{}, error) {
 	sID := p.ByName("sid")
 	if sID == "" {
 		return nil, trace.BadParameter("missing sid in request URL")
 	}
 
+	clt, err := ctx.GetUserClient(site)
```
Comment (ibeckermayer) [inline lib/web/desktop_playback.go:41]: @russjones can you take a look at my PR description and confirm that I'm correctly understanding how `ctx.GetUserClient(site)` works here?

**Review by zmb3 — APPROVED** [review]
While we're here, do you mind updating the godocs on `WithAuth` and `WithClusterAuth` so that we don't have to remember all of this again next time?

**Comment by ibeckermayer** [comment]
> While we're here, do you mind updating the godocs on `WithAuth` and `WithClusterAuth` so that we don't have to remember all of this again next time?

078b9153fe30eeb4dd9471109e5ffb6d7afc5956

**Review by zmb3 — APPROVED** [review]

**Comment by ibeckermayer** [comment]
(No need to backport to branch/v8 because desktop playback is not available)

**Pull request merged**